internal/k8sutil/k8sutil.go

```diff
@@ -150,21 +153,56 @@ func QuoteAttestationDelete(ctx context.Context, c client.Client, instanceName s
 		},
 	}
 
-	err := c.Delete(ctx, sgxAttestation)
-	if err != nil && errors.IsNotFound(err) {
-		return nil
+	if err := client.IgnoreNotFound(UnsetFinalizer(ctx, c, sgxAttestation)); err != nil {
+		return fmt.Errorf("failed unset finalizer for '%s/%s': %v", ns, instanceName, err)
 	}
-	return err
+
+	return client.IgnoreNotFound(c.Delete(ctx, sgxAttestation))
 }
 
-// Converts signer name to valid Kubernetes object name
-//  Ex:- intel.com/tcs -> tcs.intel.com
-//       tcsissuer.tcs.intel.com/sgx-ca1 -> sgx-ca1.tcsissuer.intel.tcs.com
-func SignerNameToResourceName(signerName string) string {
+// Converts signer name to valid Kubernetes object name and nanespace
+//  Ex:- intel.com/tcs -> tcs.intel.com, ""
+///      tcsissuer.tcs.intel.com/sandbox.sgx-ca -> sgx-ca.tcs.intel.com, sandbox
+//       tcsclusterissuer.tcs.intel.com/sgx-ca1 -> sgx-ca1.tcsclusterissuer.intel.tcs.com, ""
+func SignerNameToResourceNameAndNamespace(signerName string) (string, string) {
 	slices := strings.SplitN(signerName, "/", 2)
 	if len(slices) == 2 {
-		return slices[1] + "." + slices[0]
+		nameParts := strings.SplitN(slices[1], ".", 2)
+		if len(nameParts) == 2 {
```

Comment:
Is is possible to have `example.com/tcs.foo.bar`? I'm just thinking if there should be error handling or do we want to always return something?

Reply:
The `example.com/tcs.foo.bar` will result in "foo.bar.example.com"  as object name and "tcs" as its namespace, I would like to keep it as a simple function call.